pkg/resources/envoy/deployment.go

```diff
@@ -28,72 +28,90 @@ import (
-			},
-			Replicas: util.Int32Pointer(r.KafkaCluster.Spec.EnvoyConfig.GetReplicas()),
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: labelSelector,
-				},
-				Spec: corev1.PodSpec{
-					ServiceAccountName: r.KafkaCluster.Spec.EnvoyConfig.GetServiceAccount(r.KafkaCluster.Name),
-					ImagePullSecrets:   r.KafkaCluster.Spec.EnvoyConfig.GetImagePullSecrets(),
-					Tolerations:        r.KafkaCluster.Spec.EnvoyConfig.GetTolerations(),
-					NodeSelector:       r.KafkaCluster.Spec.EnvoyConfig.GetNodeSelector(),
-					Containers: []corev1.Container{
-						{
-							Name:  "envoy",
-							Image: r.KafkaCluster.Spec.EnvoyConfig.GetEnvoyImage(),
-							Ports: append(exposedPorts, []corev1.ContainerPort{
-								{Name: "envoy-admin", ContainerPort: 9901, Protocol: corev1.ProtocolTCP}}...),
-							VolumeMounts: volumeMounts,
-							Resources:    *r.KafkaCluster.Spec.EnvoyConfig.GetResources(),
-						},
-					},
-					Volumes: volumes,
+	volumes := []corev1.Volume{
+		{
+			Name: configName(envoyConfig),
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{Name: configName(envoyConfig)},
+					DefaultMode:          util.Int32Pointer(0644),
 				},
 			},
 		},
 	}
+
+	podSpec := corev1.PodSpec{
+		ServiceAccountName: kafkaCluster.Spec.EnvoyConfig.GetServiceAccount(kafkaCluster.Name),
+		ImagePullSecrets:   kafkaCluster.Spec.EnvoyConfig.GetImagePullSecrets(),
+		Tolerations:        kafkaCluster.Spec.EnvoyConfig.GetTolerations(),
+		NodeSelector:       kafkaCluster.Spec.EnvoyConfig.GetNodeSelector(),
+		Containers: []corev1.Container{
+			{
+				Name:  "envoy",
+				Image: kafkaCluster.Spec.EnvoyConfig.GetEnvoyImage(),
+				Ports: append(exposedPorts, []corev1.ContainerPort{
+					{Name: "envoy-admin", ContainerPort: 9901, Protocol: corev1.ProtocolTCP}}...),
+				VolumeMounts: volumeMounts,
+				Resources:    *kafkaCluster.Spec.EnvoyConfig.GetResources(),
+			},
+		},
+		Volumes: volumes,
+	}
+	if envoyConfig.NodeAffinity != nil {
+		podSpec.Affinity = &corev1.Affinity{
+			NodeAffinity: envoyConfig.NodeAffinity,
+		}
+	}
+	return podSpec
 }
 
-func getExposedContainerPorts(extListeners []v1beta1.ExternalListenerConfig, brokers []v1beta1.Broker) []corev1.ContainerPort {
+func getExposedContainerPorts(envoyConfig *v1beta1.EnvoyConfig, extListeners []v1beta1.ExternalListenerConfig, clusterSpec v1beta1.KafkaClusterSpec) []corev1.ContainerPort {
 	var exposedPorts []corev1.ContainerPort
 
 	for _, eListener := range extListeners {
-		for _, broker := range brokers {
-			exposedPorts = append(exposedPorts, corev1.ContainerPort{
-				Name:          fmt.Sprintf("broker-%d", broker.Id),
-				ContainerPort: eListener.ExternalStartingPort + broker.Id,
-				Protocol:      corev1.ProtocolTCP,
-			})
+		for _, broker := range clusterSpec.Brokers {
+			if envoyConfig.Id == envoyGlobal || envoyConfig.Id == broker.BrokerConfigGroup {
```

Comment:
can envoyConfig.Id be different than envoyGlobal or BrokerConfigGroup?

Reply:
This loop will iterate through all brokers. If we use 1 envoy per broker config group, we want to add to the envoy config only the brokers from the same AZ (with the envoy).
Alternatively, we should add all the brokers if we are using the `envoyGlobal`